spec/notifiers/notifier/base_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe Notifier::Base, type: :notifier do
+  let!(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    describe 'A specific notifier' do
+      class self::DummyNotifier < Notifier::Base
+        def dummy_created(actor, object, user)
+          create_activity(actor: actor, object: object, event: :created).notify(user, :popup).
+            notify(user, :email).save
+        end
+
+        def dummy_updated(actor, object, course)
+          create_activity(actor: actor, object: object, event: :created).notify(course, :feed).
+            notify(course, :email).save
+        end
+      end
+
+      let(:user) { create(:user) }
+      let(:course) { create(:course) }
+      let!(:course_users) { create(:course_user, course: course) }
+
+      context 'when notify a user' do
+        before do
+          allow_any_instance_of(self.class::DummyNotifier).to receive(:notification_view_path).
```

Comment:
that's bad too. directly create the notifier since you declared the notifier here in the spec directly?